src/python/deepgnn/graph_engine/snark/dispatcher.py

```diff
@@ -189,7 +188,7 @@ def __init__(
         num_partitions: int,
         meta: str,
         partion_func: typing.Callable[[str], int],
-        decoder_type: DecoderType,
+        decoder_class: typing.Any,
```

Comment:
why do you want to use any type, not an instance variable of subtype `Decoder`(which is in the docstring?